lib/net.js

```diff
@@ -1174,11 +1176,27 @@ function Server(options, connectionListener) {
     configurable: true, enumerable: false
   });
 
+  Object.defineProperty(this, '_handle', {
+    set: function(val) {
+      if (val === null) {
+        this[serverListenHasBeenCalled] = false;
+      }
+      this[serverHandle] = val;
+    },
+    get: function() {
+      return this[serverHandle];
+    },
+    enumerable: true
+  });
```

Comment:
I don't know how I feel about turning an internal property into an accessor...  Why did you decide on this approach?  Is it because of https://github.com/nodejs/node/pull/8419#issuecomment-251793465?